### PR TITLE
mgr/dashboard: remove cherrypy_backports.py

### PR DESCRIPTION
**Pull request by nizamial09**
since its mostly used only for older cherrypy versions which we don't
support anymore in any of our recent upstream releases, we could remove
it completely

Fixes: https://tracker.ceph.com/issues/68802
Fixes: https://tracker.ceph.com/issues/68813

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [x] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Review by epuertat — APPROVED** [review]
LGTM. Another possibility would be to deprecate this code, as those are really old versions.

**Comment by nizamial09** [comment]
> Another possibility would be to deprecate this code, as those are really old versions.

agreed! i meant this as a faster and cleaner way to solve the ongoing issue so that all the breaking test suites could be unblocked. But I agree, we can probably remove that entire code since we are using newer cherrypy in the latest version. 

I created a tracker for this now: https://tracker.ceph.com/issues/68813. cc: @afreen23 

@cbodley: could you help me schedule a QA run against this? Also is this happening only in main or its happening across all other versions too?

**Comment by nizamial09** [comment]
jenkins test docs

**Comment by nizamial09** [comment]
was able to verify this in my local dev environment.

**Review by afreen23 — APPROVED** [review]

**Comment by nizamial09** [comment]
saw some error in the docs build check.. so decided to remove this completely.

**Review by epuertat — APPROVED** [review]

File: src/pybind/mgr/prometheus/module.py [hunk 1]
```diff
@@ -29,22 +29,6 @@
...
-# exception.
-if cherrypy is not None:
-    Version = version.Version
-    v = Version(cherrypy.__version__)
-    # the issue was fixed in 3.2.3. it's present in 3.2.2 (current version on
-    # centos:7) and back to at least 3.0.0.
-    if Version("3.1.2") <= v < Version("3.2.3"):
-        # https://github.com/cherrypy/cherrypy/issues/1100
-        from cherrypy.process import servers
-        servers.wait_for_occupied_port = lambda host, port: None
```
Comment (adk3798) [inline src/pybind/mgr/prometheus/module.py:45]: some flake8 complaints about this causing make check to fail
```
prometheus/module.py:11:1: F401 'packaging.version' imported but unused
prometheus/module.py:33:1: E302 expected 2 blank lines, found 1
prometheus/module.py:991:19: F811 redefinition of unused 'version' from line 11
prometheus/module.py:1053:19: F811 redefinition of unused 'version' from line 11
prometheus/module.py:1298:27: F811 redefinition of unused 'version' from line 11
1     E302 expected 2 blank lines, found 1
1     F401 'packaging.version' imported but unused
3     F811 redefinition of unused 'version' from line 11
```

**Review by adk3798 — APPROVED** [review]
Worked for me in local testing. Just some flake8 things to cleanup.

**Comment by cbodley** [comment]
> @cbodley: could you help me schedule a QA run against this?

@nizamial09 rgw suite pending in https://pulpito.ceph.com/cbodley-2024-11-04_16:18:11-rgw-cherrypy-version-get-fix-distro-default-smithi/

> Also is this happening only in main or its happening across all other versions too?

yeah, seeing these failures on squid and reef also

**Comment by adk3798** [comment]
> > @cbodley: could you help me schedule a QA run against this?
> 
> @nizamial09 rgw suite pending in https://pulpito.ceph.com/cbodley-2024-11-04_16:18:11-rgw-cherrypy-version-get-fix-distro-default-smithi/
> 
> > Also is this happening only in main or its happening across all other versions too?
> 
> yeah, seeing these failures on squid and reef also

this also showed up in a quincy-release branch run https://pulpito.ceph.com/gabrioux-2024-11-04_07:50:02-orch:cephadm-quincy-release-distro-default-smithi/

File: src/pybind/mgr/prometheus/module.py (hunk 1 above)
Comment (nizamial09) [reply src/pybind/mgr/prometheus/module.py:45]: fixed it.

**Comment by nizamial09** [comment]
> there with references to the removed dashboard/cherrypy_backports.py:

that is weird.. is there some sort of cache here? I am not really good with the teuthology stuffs so not sure how a removed file still gets showed up in the teuthology runs? @adk3798 have you ever seen something like this before? I am checking your ongoing run and the same issue is still present in your logs as well..

**Comment by salieri11** [comment]
jenkins test dashboard cephadm

**Comment by cbodley** [comment]
> very strange.. [still failing](http://qa-proxy.ceph.com/teuthology/cbodley-2024-11-04_16:18:11-rgw-cherrypy-version-get-fix-distro-default-smithi/7980094/teuthology.log) there with references to the removed dashboard/cherrypy_backports.py:

apologies, only one job failed this way, and it was an upgrade from reef. so i'd consider rgw testing successful

**Comment by adk3798** [comment]
> > there with references to the removed dashboard/cherrypy_backports.py:
> 
> that is weird.. is there some sort of cache here? I am not really good with the teuthology stuffs so not sure how a removed file still gets showed up in the teuthology runs? @adk3798 have you ever seen something like this before? I am checking your ongoing run and the same issue is still present in your logs as well..

In the orch suite some tests use cephadm and ceph from branches rather than/as well as the build, including the test_cephadm task and all the upgrade tests. The ones using only the one from the build passed, so I think this worked

**Comment by nizamial09** [comment]
thank you for the confirmation @cbodley @adk3798. @afreen23 i am going to merge this now given the dev and test environments are broken right now.

**Pull request merged**